indico/htdocs/js/indico/modules/registration/form/tpls/fields/dialogs/size.tpl.html

```diff
@@ -9,3 +9,26 @@
         max="60">
     <i class="info-helper" title="{{ 'Must be between 5 and 60' | i18n }}"></i>
 </div>
+<ng-if="field.inputType == 'text'>
+    <div class="regFormEditLine">
+        <label class="regFormDialogCaption">
+            {{ 'Max length in char' | i18n }}
+        </label>
+        <input type="number" name="maxlength" ng-model="formData.maxLength">
+        <i class="info-helper" title="{{ 'Maximimum allowed length' | i18n }}"></i>
+    </div>
+    <div class="regFormEditLine">
+        <label class="regFormDialogCaption">
+            {{ 'Min length in char' | i18n }}
+        </label>
+        <input type="number" name="minlength" ng-model="formData.minLength">
+        <i class="info-helper" title="{{ 'Minimimum allowed length' | i18n }}"></i>
+    </div>
+    <div class="regFormEditLine">
+        <label class="regFormDialogCaption">
+            {{ 'Validation pattern' | i18n }}
+        </label>
+        <input type="text" name="validpattern" ng-model="formData.validPattern">
+        <i class="info-helper" title="{{ 'Validation pattern(a javascript regualr expression) }}' | i18n }}"></i>
```

Comment:
missing space before `(` and a typo (regualr).

Anyway, I still don't like the idea of having the pattern option. Yes, it is powerful, but eventually we want server-side validation for this stuff and except when restricting yourself to a common subset of the regex features of HTML5/JS and Python, that won't be guaranteed to work on both sides...

Also, having user-provided regexps run on the server may be risky - usually there are some ways to write regular expressions that result in [horrible performance](http://www.regular-expressions.info/catastrophic.html).

Reply:
I splitted this part out, as regexp validation is not  too important for now.